src/components/utils/popover.ts

```diff
@@ -381,30 +451,198 @@ export default class Popover extends EventsDispatcher<PopoverEvent> {
       }));
     }
 
+    if (item.isActive) {
+      el.classList.add(Popover.CSS.itemActive);
+    }
+
+    if (item.isDisabled) {
+      el.classList.add(Popover.CSS.itemDisabled);
+    }
+
     return el;
   }
 
   /**
    * Item click handler
    *
    * @param itemEl - clicked item
+   * @param event - click event
    */
-  private itemClicked(itemEl: HTMLElement): void {
-    const allItems = this.nodes.wrapper.querySelectorAll(`.${Popover.CSS.item}`);
-    const itemIndex = Array.from(allItems).indexOf(itemEl);
+  private itemClicked(itemEl: HTMLElement, event: PointerEvent): void {
+    const allItems = Array.from(this.nodes.items.children);
+    const itemIndex = allItems.indexOf(itemEl);
     const clickedItem = this.items[itemIndex];
 
-    clickedItem.onClick(clickedItem);
+    if (clickedItem.isDisabled) {
+      return;
+    }
+
+    /**
+     * If there is any other item in confirmation state except the clicked one, clean it up
+     */
+    allItems
+      .filter(item => item !== itemEl)
+      .forEach(item => {
+        this.cleanUpConfirmationStateForItem(item);
+      });
+
+    if (clickedItem.confirmation) {
+      this.enableConfirmationStateForItem(clickedItem as PopoverItemWithConfirmation, itemEl, itemIndex);
+
+      return;
+    }
+    clickedItem.onActivate(clickedItem, event);
+
+    if (this.shouldHighlightActivated) {
+      /** If popover is not intended to close once item is activated, we need to update active state of activated item */
+      clickedItem.isActive = !clickedItem.isActive;
+      itemEl.classList.toggle(Popover.CSS.itemActive);
+    }
+
+    if (clickedItem.closeOnActivate) {
+      this.hide();
+    }
+  }
+
+  /**
+   * Enables confirmation state for specified item.
+   * Replaces item element in popover so that is becomes highlighted in a special way
+   *
+   * @param item - item to enable confirmation state for
+   * @param itemEl - html element corresponding to the item
+   * @param itemIndex - index of the item in all items list
+   */
+  private enableConfirmationStateForItem(item: PopoverItemWithConfirmation, itemEl: HTMLElement, itemIndex: number): void {
+    /** Save root item requiring confirmation to restore original state on popover hide */
+    if (this.itemsRequiringConfirmation[itemIndex] === undefined) {
+      this.itemsRequiringConfirmation[itemIndex] = item;
+    }
+    const newItemData = {
+      ...item,
+      ...item.confirmation,
+      confirmation: item.confirmation.confirmation,
+    } as PopoverItem;
+
+    this.items[itemIndex] = newItemData;
+
+    const confirmationStateItemEl = this.createItem(newItemData as PopoverItem);
+
+    confirmationStateItemEl.classList.add(Popover.CSS.itemConfirmation, ...Array.from(itemEl.classList));
```

Comment:
Wouldn't it work?
```suggestion
    confirmationStateItemEl.classList.add(Popover.CSS.itemConfirmation, ...itemEl.classList);
```

Reply:
nope 
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/31101125/198852747-d19cc520-719d-4724-9694-46c8e9ee33cc.png">